### DISCUSSION TITLE
How to define a specific error message when checkIfExists of fromPath is failed

### DISCUSSION THREAD

**Discussion by liuyang2006** [nextflow-io/nextflow#2515] [Q&A] [answered]
Dear Nextflow,
I want my pipeline show a specific error message that can help the users when the statement is not passed: `genome_ch = Channel.fromPath(genome_path, type: 'any', checkIfExists: true)` , however, I could not found how to achieve this. Would you please help me to find a way? Thank you very much!

Running command:
```
nextflow run TheJacksonLaboratory/nanome \
      --input https://github.com/TheJacksonLaboratory/nanome/raw/master/test_data/ecoli_ci_test_fast5.tar.gz\
       --dsname Test --genome hg1234
```

Error message now:
```
No such file: /gpfs/ctgs0/fastscratch/liuya/nanome/test/hg1234
```

I want to change the error message to my specific message in order to tell users more information. Thank you!

**Comment by manuelesimi** [comment, marked as answer]
Hello,

You could check yourself if the path exists and return a custom message with the following code:

```
genome_path = file(genome_path)
if ( !genome_path.exists() )
    exit 1, "Genome path does not exist"

genome_ch = Channel.fromPath(genome_path, type: 'any')
```
Thanks,
manuele

**Comment by liuyang2006** [comment]
Thank you! This solution can solve my problem.
